modules/@apostrophecms/piece-type/index.js

```diff
@@ -178,8 +178,8 @@ module.exports = {
       return result;
     },
     async getOne(req, _id) {
-      self.publicApiCheck(req);
       _id = self.inferIdLocaleAndMode(req, _id);
+      self.publicApiCheck(req);
```

Comment:
Why move these?

Reply:
Inferring the mode alters the result of the permissions check made in publicApiCheck, allowing contributors to do things they should be able to do without forcing us to change our REST API design to always require ?apos-mode even when it can be inferred from the _id.